src/diffusers/models/modeling_utils.py

```diff
@@ -179,22 +182,55 @@ def is_gradient_checkpointing(self) -> bool:
         """
         return any(hasattr(m, "gradient_checkpointing") and m.gradient_checkpointing for m in self.modules())
 
-    def enable_gradient_checkpointing(self) -> None:
+    def enable_gradient_checkpointing(
+        self,
+        gradient_checkpointing_func: Optional[Callable] = None,
+        gradient_checkpointing_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
```

Comment:
can you give an example when they need to pass both func and kwargs? can't we just accept a  `gradient_checkpointing_func`?

Reply:
ok looks good to me now :)